### PR TITLE
jwk #2: ensure jwk txns are expected in consensus

### PR DESCRIPTION
**Pull request by zjma**

### CODE REVIEW

**Review by grao1991 — APPROVED** [review]

File: consensus/safety-rules/src/safety_rules_manager.rs [hunk 1]
```diff
@@ -73,6 +76,17 @@ pub fn storage(config: &SafetyRulesConfig) -> PersistentSafetyStorage {
     }
 }
 
+pub fn load_consensus_key_from_secure_storage(
```
Comment (zekun000) [inline consensus/safety-rules/src/safety_rules_manager.rs:79]: we already have something for DKG?

File: consensus/src/util/mod.rs [hunk 2]
```diff
@@ -2,7 +2,24 @@
...
+};
+
 pub mod db_tool;
 #[cfg(any(test, feature = "fuzzing"))]
 pub mod mock_time_service;
 pub mod time_service;
+
+pub fn is_vtxn_expected(features: &Features, vtxn: &ValidatorTransaction) -> bool {
+    match vtxn {
+        ValidatorTransaction::DummyTopic1(_) | ValidatorTransaction::DummyTopic2(_) => true,
```
Comment (zekun000) [inline consensus/src/util/mod.rs:17]: this shouldn't be true, it opens ddos attacks by validators? honestly I don't think it should even exist on non-testing path

Comment (zjma) [reply consensus/src/util/mod.rs:17]: once the corresponding feature is enabled, DKG/JWK txn are similar?

Comment (zekun000) [reply consensus/src/util/mod.rs:17]: DKG/JWK has verification logic that only valid ones can go through? the dummy txn execution just succeeds?

**Review by danielxiangzl — COMMENTED** [review]
Can approve once dummy vtxn is marked test-only.

File: consensus/src/util/mod.rs (hunk 2 above)
Comment (zjma) [reply consensus/src/util/mod.rs:17]: fixed

**Comment by zjma** [comment]
> Can approve once dummy vtxn is marked test-only.

fixed

**Review by danielxiangzl — APPROVED** [review]

**Pull request merged**